Mathlib/Geometry/Manifold/IntegralCurve/Basic.lean

```diff
@@ -0,0 +1,164 @@
+/-
+Copyright (c) 2024 Winston Yin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Winston Yin
+-/
+import Mathlib.Geometry.Manifold.MFDeriv.Tangent
+
+/-!
+# Integral curves of vector fields on a manifold
+
+Let `M` be a manifold and `v : (x : M) → TangentSpace I x` be a vector field on `M`. An integral
+curve of `v` is a function `γ : ℝ → M` such that the derivative of `γ` at `t` equals `v (γ t)`. The
+integral curve may only be defined for all `t` within some subset of `ℝ`.
+
+## Main definitions
+
+Let `v : M → TM` be a vector field on `M`, and let `γ : ℝ → M`.
+* `IsIntegralCurve γ v`: `γ t` is tangent to `v (γ t)` for all `t : ℝ`. That is, `γ` is a global
+integral curve of `v`.
+* `IsIntegralCurveOn γ v s`: `γ t` is tangent to `v (γ t)` for all `t ∈ s`, where `s : Set ℝ`.
+* `IsIntegralCurveAt γ v t₀`: `γ t` is tangent to `v (γ t)` for all `t` in some open interval
+around `t₀`. That is, `γ` is a local integral curve of `v`.
+
+For `IsIntegralCurveOn γ v s` and `IsIntegralCurveAt γ v t₀`, even though `γ` is defined for all
+time, its value outside of the set `s` or a small interval around `t₀` is irrelevant and considered
+junk.
+
```

Comment:
Can you mention the other files as well? Either as a reference (this file proves xxx, the other files prove yyy), or even a detailed list. ("The following properties are shown in `xxx`." The module doc-string can be helpful for discovering results, so referring to other files is fine and can be beneficial!)